### ISSUE TITLE
promises or callbacks

### ISSUE THREAD

**Issue by Raynos** [pillarjs/discussions#11]
Various modules support callbacks or promises.

finalhandler is callback, csrf is both and templation is promises.

I'd be nice if jshttp & pillarjs consistently support either all promises, all callbacks or all both.

I'd personally recommend that we try to support both, i.e. you SHOULD support both and MUST support callbacks.

**Comment by dougwilson** [comment]
Yes, I've has some discussions about it with @jonathanong . `finalhandler` itself is probably an exception, because I don't see how it could possibly even use promises. It would, really. It's not even a callback-style, because you're not passing in a callback to it at all. You can use it with promises and your `catch` function.

**Comment by Fishrock123** [comment]
> I'd personally recommend that we try to support both, i.e. you SHOULD support both and MUST support callbacks.

So long as we don't depend on promise libs imo..
